examples/simple-chat/src/simple_chat.ts

```diff
@@ -60,17 +60,22 @@ class ChatUI {
     // When we detect low maxStorageBufferBindingSize, we assume that the device (e.g. an Android
     // phone) can only handle small models and make all other models unselectable. Otherwise, the
     // browser may crash. See https://github.com/mlc-ai/web-llm/issues/209.
+    // Also use GPU vendor to decide whether it is a mobile device (hence with limited resources).
+    const androidMaxStorageBufferBindingSize = 1 << 27;  // 128MB
+    const mobileVendors = new Set<string>(["qualcomm", "arm"])
     let restrictModels = false;
     let maxStorageBufferBindingSize: number;
+    let gpuVendor: string;
     try {
       maxStorageBufferBindingSize = await chat.getMaxStorageBufferBindingSize();
+      gpuVendor = await chat.getGPUVendor();
```

Comment:
@CharlieFRuan To speed things up, you can do the following:

```suggestion
      [maxStorageBufferBindingSize, gpuVendor] = await Promise.all([
        chat.getMaxStorageBufferBindingSize(),
        chat.getGPUVendor(),
      ]);
```